src/settings.js

```diff
@@ -583,12 +583,15 @@ var ASYNCIFY_WHITELIST = ['qsort',
 // given that EXTRA_EXPORTED_RUNTIME_METHODS exists, and so this option exists
 // only for backwards compatibility. You should use
 // EXTRA_EXPORTED_RUNTIME_METHODS for things you want to export from the
-// runtime.  Note that methods on this list are only exported if they are
-// included (either automatically from linking, or due to being in
-// DEFAULT_LIBRARY_FUNCS_TO_INCLUDE).
-// Note that the name may be slightly misleading, as this is for any JS library
-// element, and not just methods. For example, we export the Runtime object by
+// runtime.
+// Methods on this list are automatically included from JS libraries, if they
+// are library methods.
```

Comment:
Does `EXPORTED_RUNTIME_METHODS` now behave like `EXPORTED_JS_FUNCTIONS` that I was referring to in the earlier comment? I.e. if I add e.g. `-s EXPORTED_RUNTIME_METHODS=['emscripten_get_device_pixel_ratio']` and there's no other C/C++/JSlib code that would reference `emscripten_get_device_pixel_ratio()`, will the function be linked in and exported to the build?

If so, then that'd actually be great, though it does not feel good to advertise this feature on `-s EXPORTED_RUNTIME_METHODS`, but rather just silently/unadvertisingly do this to support `ccall` for backwards compatibility? And then add a new `EXPORTED_JS_FUNCTIONS` that does this same, and advertise and document it instead? That way we can move a little bit towards getting rid of the "runtime methods" concept altogether?

(sorry if I misunderstood something in this comment?)

Reply:
Yeah, I think your first paragraph describes how it works.

There's no special-casing of ccall here? Maybe I am misreading the second comment?

In any case, I agree with your general point - we should simplify and unify this stuff. I think it may be best to close this PR, as I don't yet have a clear picture in my head of how to make proper progress here, and the immediate ccall code size problem has been fixed in that other PR.